### PR TITLE
[Ambient] Validate L7 Istio Configurations 

### LINKED ISSUE #7900: [ambient] Disable L7 istio configs for services that are in Ambient but doesn't have a waypoint proxy
### What do you want to improve?
If a service is in Ambient but just with L4 processing, it shouldn't let the user create L7 traffic policies, as they don't be applicable:
image
t might confuse the user if Kiali let them create the istio config but then it is not taking effect.

### What is the current behavior?
If a service is in Ambient with just L4 processing, Kiali enables L7 istio configs.

### What is the new behavior?
If the service is in an Ambient Mesh, but it doesn't have a waypoint, should disable L7 istio configs not applicable. I think just the TCP Traffic shifting should be enabled:

image

Ref. https://github.com/kiali/kiali/issues/7826

### PR DESCRIPTION
**Pull request by josunect**
### Describe the change
Adds Ambient L7 Istio config validations so Kiali warns when policies need a waypoint but won’t take effect (validate, don’t block wizards).

- Shared L4/L7 classification helpers (business/checkers/ambient/l7_classification.go), reused by MCP analyze_ambient_policies
- New AmbientPolicyChecker with codes for:
  - AuthPolicy / RequestAuthentication: missing enrollment and missing targetRefs (KIA0109/KIA0110, KIA1110/KIA1115)
  - VS/DR: destination not enrolled, CR not in service namespace (KIA1113/KIA1114, KIA0211/KIA0212)
  - WasmPlugin / Telemetry L7 without enrollment (KIA1111/KIA1112)
  - Workload L7-only AuthPolicy without waypoint (KIA1317)
- VS/DR host path highlighting (spec/hosts[i], spec/host)
- Fix: validation criteria now includes WasmPlugins and Telemetry so those checks actually run

image

Out of scope
- AuthPolicy / RA / Wasm / Telemetry in a different namespace than Ambient workloads (skipped when CR ns is not Ambient; VS/DR cross-ns is covered)
- Validating targetRefs quality (existence only; no missing Service, use-waypoint=none on target, etc.)
- Gateway API routes (HTTPRoute / GRPCRoute / …), including DR subsets → version-specific Services for HTTPRoute
- Blocking L7 wizards / create path
- Root-namespace mesh-wide policies beyond skipping control-plane false positives (e.g. GatewayClass targetRefs) and multi-cluster Ambient edge cases
- EnvoyFilter (not supported on waypoints; no Ambient validation)

### Steps to test the PR

- Install Istio Ambient
- See https://github.com/user-attachments/files/30127656/ambient-l7-manual-validation.md for different test cases

### Automation testing

If applicable, explain the case scencarios covered by **unit / integration / e2e** tests created for this PR.

### Issue reference
Fixes #7900

### CODE REVIEW

**Comment by josunect** [comment]
Some manual testing done: 

[ambient-l7-manual-validation.md](https://github.com/user-attachments/files/30127656/ambient-l7-manual-validation.md)

Not comited, but can be useful for manual validation, examples, and issue reproduction notes.

e2e tests not added because they take a lot of time running and the logic is validated in unit tests.

**Comment by josunect** [comment]
@coderabbitai review

**Comment by josunect** [comment]
@coderabbitai review

File: business/checkers/ambient/ambient_workload_checker.go [hunk 1]
```diff
@@ -127,8 +127,15 @@ func (awc AmbientWorkloadChecker) hasAuthPolicyAndNoWaypoint() bool {
...
+	}
 	for _, ap := range awc.authorizationPolicies {
-		if ap.Namespace == awc.workload.Namespace && len(awc.workload.WaypointWorkloads) == 0 {
+		if ap.Namespace != awc.workload.Namespace {
+			continue
+		}
+		// Only L7 AuthorizationPolicies require a waypoint; L4 policies are enforced by ztunnel.
+		if isL7, _ := IsL7AuthorizationPolicy(&ap.Spec); isL7 {
 			return true
 		}
```
Comment (hhovsepy) [inline business/checkers/ambient/ambient_workload_checker.go:156]: **High priority — KIA1317 fan-out / false positives**

`hasAuthPolicyAndNoWaypoint` returns true when *any* L7 AuthorizationPolicy exists in the workload namespace, without checking whether that policy actually targets this workload (selector match or `targetRefs` → Service/Gateway that covers it).

In practice this flags every Ambient workload without a waypoint in the ns (e.g. `productpage`, `details`, traffic-gen) whenever a single L7 AuthPolicy is present — even if the policy only selects `app: reviews`.

Ambient attachment is scoped; suggest narrowing to workloads that the L7 policy actually applies to (selector match and/or services referenced by `targetRefs`), similar to how VS/DR warnings are destination-scoped.

Verified live: with `kia1317-reviews-l7-deny-get` selecting `app: reviews`, KIA1317 also appeared on unrelated bookinfo workloads.

File: business/checkers/ambient_policy_checker.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+		key, validation := EmptyValidValidation(wp.Name, wp.Namespace, kubernetes.WasmPlugins, c.Cluster)
+		matchLabels := map[string]string{}
+		if wp.Spec.Selector != nil {
+			matchLabels = wp.Spec.Selector.MatchLabels
+		}
+		if c.selectorNeedsWarning(matchLabels, wp.Namespace, nsByName[wp.Namespace], nsStatus) {
+			check := models.Build("wasmplugin.ambient.l7nowaypoint", "")
+			validation.Checks = append(validation.Checks, &check)
+			validation.Valid = false
+		}
```
Comment (hhovsepy) [inline business/checkers/ambient_policy_checker.go:162]: **High priority — WasmPlugin missing `targetRefs` parity**

AuthPolicy / RequestAuthentication correctly warn when waypoints would ignore selector-only attachment (`l7notargetrefs` → KIA0110 / KIA1115).

WasmPlugin (and optionally L7 Telemetry) only check enrollment via `selectorNeedsWarning` (KIA1111), but Istio Ambient requires `targetRefs` for waypoint attachment as well — selector-based Wasm is ignored by waypoints ([L7 features](https://istio.io/latest/docs/ambient/usage/l7-features/), WasmPlugin `targetRefs` note).

Suggest adding the same pattern as RA/AuthPolicy here, e.g. warn when Ambient + missing `targetRefs` (new code or reuse a shared message), in addition to the enrollment warning.

Verified live: selector-only WasmPlugin in Ambient without enrollment → KIA1111 only; no `targetRefs` warning.

**Review by hhovsepy — COMMENTED** [review]
## Ambient validation review (logic follow-ups)

Reproduced the new Ambient L7 codes against a live Ambient bookinfo cluster. Overall the checks match Ambient semantics and **warning** severity looks right (including dual AuthPolicy/RA warnings and selector/`targetRefs` as warning).

Two higher-priority follow-ups below (inline). Neither is a merge blocker for the Ambient story, but they are the main correctness/parity gaps left.

Thanks for the work on this — the VS/DR destination-service enrollment model and AuthPolicy/RA `targetRefs` checks look solid.

**Comment by hhovsepy** [comment]
Hi @josunect , very nice work, still testing locally, seems all Warnings are shown fine.
I asked Cursor to put some higher priority points as comments.

**Comment by hhovsepy** [comment]
One question @josunect 
is this error KIA1317 expected in sidecar mode?
image

**Comment by josunect** [comment]
> One question @josunect is this error KIA1317 expected in sidecar mode? image

No, this shouldn't be a warning in sidecars, I'll check

**Comment by josunect** [comment]
@hhovsepy do you have the steps to reproduce it? I'm not able to:

image

**Comment by hhovsepy** [comment]
@josunect Yes — here are the repro steps for **KIA1317 on sidecar / non-Ambient**, using `kia1317-reviews-l7-deny-get`.

### Env
- Sidecar mesh (not Ambient): `bookinfo` has `istio-injection=enabled`, **no** `istio.io/dataplane-mode=ambient`
- Kiali reports `bookinfo.isAmbient == false`

### Apply (minimal)
```bash
kubectl apply -f - <<'YAML'
apiVersion: security.istio.io/v1
kind: AuthorizationPolicy
metadata:
  name: kia1317-reviews-l7-deny-get
  namespace: bookinfo
spec:
  action: DENY
  selector:
    matchLabels:
      app: reviews
  rules:
  - to:
    - operation:
        methods: ["GET"]
YAML
```

### Where to look
Workloads → `reviews-v1` / `reviews-v2` / `reviews-v3` (not the AuthPolicy object).

You should see **KIA1317**: *This workload has L7 Authorization Policies but no Waypoint*.

### Notes
1. In sidecar mode this warning is incorrect — L7 AuthPolicies with selectors are normal and do not need a waypoint.
2. If other L7 APs that also select `app: reviews` (or `targetRefs` → Service `reviews`) are present, they will also trigger KIA1317 on the same workloads. For a clean repro, keep only `kia1317-reviews-l7-deny-get`.
3. AuthPolicy Ambient codes (KIA0109/0110) correctly stay quiet in non-Ambient; only **KIA1317** leaks.

### Suggested fix
Gate `hasAuthPolicyAndNoWaypoint` on Ambient namespace (same pattern as Ambient policy checkers):

```go
func (awc AmbientWorkloadChecker) hasAuthPolicyAndNoWaypoint() bool {
	// KIA1317 is Ambient-only: in sidecar mode L7 AuthPolicies apply via selectors without a waypoint.
	if !awc.namespaces.IsNamespaceAmbient(awc.workload.Namespace, awc.cluster) {
		return false
	}
	// ... existing checks ...
}
```

Opening a small PR against `issue7900` with this change + unit test next.

**Comment by hhovsepy** [comment]
Follow-up: opened a small fix PR against `issue7900` with the Ambient gate + unit test:

https://github.com/josunect/kiali/pull/32

**Comment by josunect** [comment]
@hhovsepy thanks, I've made a review and complete the check in other policies as well

**Review by hhovsepy — APPROVED** [review]
Verified, thanks @josunect all works fine.

File: business/checkers/ambient/ambient_workload_checker.go [hunk 3]
```diff
@@ -127,11 +134,100 @@ func (awc AmbientWorkloadChecker) hasAuthPolicyAndNoWaypoint() bool {
...
+			continue
+		}
+		if labels.SelectorFromSet(svc.Spec.Selector).Matches(labels.Set(awc.workload.Labels)) {
+			return true
+		}
+	}
+
+	// Fallback when Services were not provided to the checker (e.g. some list paths).
+	for _, svc := range awc.workload.Services {
+		if svc.Name == serviceName {
```
Comment (jshaughn) [inline business/checkers/ambient/ambient_workload_checker.go:226]: Should this be namespace constrained?

**Review by jshaughn — COMMENTED** [review]
@josunect , @hhovsepy , thanks for this work. Sorry for not getting time to review/test. A short post-merge review looks good to me. I did add one minor comment.

File: business/checkers/ambient/ambient_workload_checker.go (hunk 3 above)
Comment (josunect) [reply business/checkers/ambient/ambient_workload_checker.go:226]: Makes sense, I have created this PR as a follow up: https://github.com/kiali/kiali/pull/10120
